src/libaktualizr/package_manager/ostreemanager.cc

```diff
@@ -19,47 +19,27 @@
 using OstreeProgressPtr = std::unique_ptr<OstreeAsyncProgress, GObjectFinalizer<OstreeAsyncProgress>>;
 
 static void aktualizr_progress_cb(OstreeAsyncProgress *progress, gpointer data) {
-  auto *percent_complete = static_cast<guint *>(data);
-
-  g_autofree char *status = ostree_async_progress_get_status(progress);
-  guint scanning = ostree_async_progress_get_uint(progress, "scanning");
-  guint outstanding_fetches = ostree_async_progress_get_uint(progress, "outstanding-fetches");
-  guint outstanding_metadata_fetches = ostree_async_progress_get_uint(progress, "outstanding-metadata-fetches");
-  guint outstanding_writes = ostree_async_progress_get_uint(progress, "outstanding-writes");
-  guint n_scanned_metadata = ostree_async_progress_get_uint(progress, "scanned-metadata");
-
-  if (status != nullptr && *status != '\0') {
-    LOG_INFO << "ostree-pull: " << status;
-  } else if (outstanding_fetches != 0) {
-    guint fetched = ostree_async_progress_get_uint(progress, "fetched");
-    guint metadata_fetched = ostree_async_progress_get_uint(progress, "metadata-fetched");
-    guint requested = ostree_async_progress_get_uint(progress, "requested");
-    if (scanning != 0 || outstanding_metadata_fetches != 0) {
-      LOG_INFO << "ostree-pull: Receiving metadata objects: " << metadata_fetched
-               << " outstanding: " << outstanding_metadata_fetches;
-    } else {
-      guint calculated = (fetched * 100) / requested;
-      if (calculated != *percent_complete) {
-        LOG_INFO << "ostree-pull: Receiving objects: " << calculated << "% ";
-        *percent_complete = calculated;
-      }
-    }
-  } else if (outstanding_writes != 0) {
-    LOG_INFO << "ostree-pull: Writing objects: " << outstanding_writes;
+  auto *mt = static_cast<PullMetaStruct *>(data);
```

Comment:
In the worst case, can't we use this `calculated`? That seems to be good enough for what we need.